standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1280,6 +1284,94 @@ public ColumnStatistics getTableStats(final String catName, final String dbName,
     return result;
   }
 
+  public List<HiveObjectPrivilege> getTableAllColumnGrants(String catName, String dbName,
+                                                           String tableName, String authorizer) throws MetaException {
+    Query query = null;
+
+    // These constants should match the SELECT clause of the query.
+    final int authorizerIndex = 0;
+    final int columnNameIndex = 1;
+    final int createTimeIndex = 2;
+    final int grantOptionIndex = 3;
+    final int grantorIndex = 4;
+    final int grantorTypeIndex = 5;
+    final int principalNameIndex = 6;
+    final int principalTypeIndex = 7;
+    final int privilegeIndex = 8;
+
+    // Retrieve the privileges from the object store. Just grab only the required fields.
+    String queryText = "select " +
+            TBL_COL_PRIVS + ".\"AUTHORIZER\", " +
+            TBL_COL_PRIVS + ".\"COLUMN_NAME\", " +
+            TBL_COL_PRIVS + ".\"CREATE_TIME\", " +
+            TBL_COL_PRIVS + ".\"GRANT_OPTION\", " +
+            TBL_COL_PRIVS + ".\"GRANTOR\", " +
```

Comment:
JDO uses LEFT OUTER JOIN In the corresponding query fired through JDO. So, I used just that. But I think LEFT OUTER JOIN doesn't have any effect here because of the WHERE clause at the end. The WHERE clauses are on the RIGHT side columns. These clauses will evaluate to false when the RIGHT side columns will be NULL, effectively turning this OUTER join into an INNER join.

It's better to leave LEFT JOIN as is in case someone wants to compare this query with JDO query. But if you think we should use INNER JOIN here, I will use INNER JOIN.

Reply:
i am not sure if the optimizer will change it to an inner join internally as there is a foreign key constraint from TBL_COL_PRIVS to TBLS ..but usually outer joins are costlier than inner joins.